### PR TITLE
config dump, listeners are added to warming list if workers are not started yet

### PR DESCRIPTION
**Pull request by incfly**
Signed-off-by: Jianfei Hu <jianfeih@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: Listeners config dump changed to warming_listeners if the workers are not started.
Risk Level: Low
Testing: Unit test, (existing unit test is good enough, with workers started before and after)
Docs Changes:
Release Notes: Listeners config dump changed to warming_listeners if the workers are not started.
[Optional Fixes #Issue] https://github.com/envoyproxy/envoy/issues/6904
[Optional Deprecated:]

### CODE REVIEW

**Comment by silentdai** [comment]
Make perfect sense. Thank you!
/lgtm

**Comment by asraa** [comment]
This looks good to me. 

Mind if you add in 
`EXPECT_EQ(getListenersConfigDump().dynamic_warming_listeners().size(), 1) `
here in ads_integration_test to just note that the pre-server initialization listeners are added as active but dumped as warming?

https://github.com/envoyproxy/envoy/blob/79b071cc1b6bf877c7c97a6e79b5be8e9b6ace38/test/integration/ads_integration_test.cc#L968-L971

**Comment by mattklein123** [comment]
Waiting on @asraa comment.

/wait

File: test/integration/ads_integration_test.cc [hunk 1]
```diff
@@ -969,6 +969,9 @@ TEST_P(AdsIntegrationTest, ListenerDrainBeforeServerStart) {
       Config::TypeUrl::get().Listener, {buildListener("listener_0", "route_config_0")},
       {buildListener("listener_0", "route_config_0")}, {}, "1");
   test_server_->waitForGaugeGe("listener_manager.total_listeners_active", 1);
+  // Before server is started, even though listeners are added to active list
+  // we mark them as "warming" in config dump since they're not initialized yet.
+  EXPECT_EQ(getListenersConfigDump().dynamic_warming_listeners().size(), 1);
```
Comment (incfly) [inline test/integration/ads_integration_test.cc]: @asraa Done.

Comment (asraa) [reply test/integration/ads_integration_test.cc]: Thank you!

**Comment by incfly** [comment]
Tests now all passed. @mattklein123 could you take another look or this is ready to go?

**Pull request merged**